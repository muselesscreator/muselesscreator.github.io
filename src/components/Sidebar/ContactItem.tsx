export type ContactItemProps = {
  IconEl: React.FC;
  title: string;
  href: string;
  value: string;
};
const ContactItem = ({
  IconEl,
  title,
  href,
  value
}: ContactItemProps) => (
  <li className="contact-item">
    <div className="icon-box">
      <IconEl />
    </div>
    <div className="contact-info">
      <p className="contact-title">{title}</p>
      <a href={href} className="contact-link">{value}</a>
    </div>
  </li>
);

export default ContactItem;
