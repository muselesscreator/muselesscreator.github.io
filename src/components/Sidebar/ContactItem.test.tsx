import {
  test,
  describe,
  expect,
} from 'vitest';

import { shallow } from '@muselesscreator/react-shallow-snapshot';

import ContactItem, { ContactItemProps } from './ContactItem';

const testIcon = 'test-icon';
const props = {
  IconEl: (() => testIcon) as React.FC,
  title: 'test-title',
  href: 'test-href',
  value: 'test-value',
} as ContactItemProps;

describe('ContactItem', () => {
  test('renders correctly', () => {
    const el = shallow(<ContactItem {...props} />);
    expect(el.snapshot).toMatchSnapshot();
    console.log({ el });
    const [iconBox, contactInfo] = el.instance.children;
    expect(iconBox.children).toHaveLength(1);
    console.log({ iconBoxChildren: iconBox.children });
    expect(iconBox.children[0].el).toEqual(testIcon);
    expect(contactInfo.children).toHaveLength(2);
    expect(contactInfo.children[0].children[0].el).toEqual(props.title);
    expect(contactInfo.children[1].children[0].el).toEqual(props.value);
    expect(contactInfo.children[1].props.href).toEqual(props.href);
  });
});
