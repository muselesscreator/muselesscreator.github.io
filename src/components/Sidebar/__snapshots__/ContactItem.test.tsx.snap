// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContactItem > renders correctly 1`] = `
"<li
  className="contact-item"
>
  <div
    className="icon-box"
  >
    test-icon
  </div>
  <div
    className="contact-info"
  >
    <p
      className="contact-title"
    >
      test-title
    </p>
    <a
      className="contact-link"
      href="test-href"
    >
      test-value
    </a>
  </div>
</li>"
`;
